sphinxgallery/gen_rst.py

```diff
@@ -444,7 +458,12 @@ def generate_file_rst(fname, target_dir, src_dir):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    exec_script = fname.startswith('plot')
+    use_mayavi = gallery_conf.get('use_mayavi', False)
+    if not use_mayavi and any('mayavi' in bl[1] for bl in script_blocks):
```

Comment:
> how do you prevent the mayavi example to crash on travis if mayavi is not installed?

Good point. I would say in a hacky way, like putting in a folder on its own and changing conf.py not to include this folder include it, renaming plot mayavi example or even deleting it. Maybe I would favor the second option because as Gaël was saying it is likely that sphinx-gallery will be copied an pasted around so the less hacks it has the better.